### PR TITLE
[WIP] Start of api for accepting invite.

### PR DESCRIPTION
**Pull request by ashimali**
This only handles the case of an existing user being invited and added to an existing service.

### CODE REVIEW

File: app/service/rest.py [hunk 1]
```diff
@@ -155,5 +155,23 @@ def get_users_for_service(service_id):
     return jsonify(data=result.data)
 
 
+@service.route('/<service_id>/users/<user_id>', methods=['POST'])
+def add_user_to_service(service_id, user_id):
+    service = dao_fetch_service_by_id(service_id)
+    if not service:
+        return _service_not_found(service_id)
+    user = get_model_users(user_id=user_id)
+
```
Comment (servingUpAces) [inline app/service/rest.py]: get returns None  - should add a check here.

File: app/service/rest.py [hunk 2]
```diff
@@ -155,5 +155,23 @@ def get_users_for_service(service_id):
...
+    if user in service.users:
+        return jsonify(result='error',
+                       message='User id: {} already part of service id: {}'.format(user_id, service_id)), 400
+
+    service.users.append(user)
+    dao_update_service(service)
```
Comment (servingUpAces) [inline app/service/rest.py]: dao_add_user_to_service does the same thing

File: tests/app/service/test_rest.py [hunk 3]
```diff
@@ -456,3 +456,127 @@ def test_default_permissions_are_added_for_user_service(notify_api,
...
+            resp = client.get(
+                '/service/{}/users'.format(sample_service.id),
+                headers=[('Content-Type', 'application/json'), auth_header]
+            )
+
+            assert resp.status_code == 200
+            result = json.loads(resp.get_data(as_text=True))
+            assert len(result['data']) == 1
+            assert result['data'][0]['email_address'] == user_already_in_service.email_address
+
```
Comment (servingUpAces) [inline tests/app/service/test_rest.py]: Why call the get first here?

**Comment by ashimali** [comment]
Updated as per comments above.

File: tests/app/service/test_rest.py (hunk 3 above)
Comment (ashimali) [inline tests/app/service/test_rest.py]: Verify only one user on service. Call post to add, verify 2 users on service, the original one and the one added.

**Pull request merged**